Source/SwiftLintFramework/Rules/VerticalWhitespaceBetweenCasesRule.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+import SourceKittenFramework
+
+private extension File {
+    func violatingRanges(for pattern: String) -> [NSRange] {
+        return match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
+    }
+}
+
+private extension Dictionary where Key: StringProtocol, Value: StringProtocol {
+    func cleanedKeysDict() -> [String: String] {
+        var cleanDict = [String: String]()
+
+        for (key, value) in self {
+            guard let keyString = key as? String else { continue }
+            let cleanedKey = keyString.replacingOccurrences(of: "↓", with: "")
+            cleanDict[cleanedKey] = value as? String
+        }
+
+        return cleanDict
+    }
+}
+
+public struct VerticalWhitespaceBetweenCasesRule: ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    private static let nonTriggeringExamples = [
+        """
+        switch x {
+
+        case 0..<5:
+            print("x is low")
+
+        case 5..<10:
+            print("x is high")
+
+        default:
+            print("x is invalid")
+
+        }
+        """,
+        """
+        switch x {
+        case 0..<5:
+            print("x is low")
+
+        case 5..<10:
+            print("x is high")
+
+        default:
+            print("x is invalid")
+        }
+        """,
+        """
+        switch x {
+        case 0..<5: print("x is low")
+        case 5..<10: print("x is high")
+        default: print("x is invalid")
+        }
+        """
+    ]
+
+    private static let violatingToValidExamples: [String: String] = [
+    """
+        switch x {
+        case 0..<5:
+            print("x is valid")
+    ↓    default:
+            print("x is invalid")
+        }
+    """: """
+        switch x {
+        case 0..<5:
+            print("x is valid")
+
+        default:
+            print("x is invalid")
+        }
+    """
+    ]
+
+    private let pattern = "([^\\n{][ \\t]*\\n)([ \\t]*(?:case[^\\n]+|default):[ \\t]*\\n)"
+}
+
+extension VerticalWhitespaceBetweenCasesRule: OptInRule, AutomaticTestableRule {
+    public var configurationDescription: String { return "N/A" }
+
+    public init(configuration: Any) throws {}
+
+    public static let description = RuleDescription(
+        identifier: "vertical_whitespace_between_cases",
+        name: "Vertical Whitespace Between Cases",
+        description: "Include a vertical whitespace (empty line) between cases in switch statements.",
+        kind: .style,
+        nonTriggeringExamples: Array(Set(violatingToValidExamples.values)) + nonTriggeringExamples,
+        triggeringExamples: Array(Set(violatingToValidExamples.keys)),
+        corrections: violatingToValidExamples.cleanedKeysDict()
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let patternRegex: NSRegularExpression = regex(pattern)
+
+        return file.violatingRanges(for: pattern).map { violationRange in
```

Comment:
this should be done using the AST instead of a regex. You can check `NoFallthroughOnlyRule` which does something similar and the discussion on the PR: https://github.com/realm/SwiftLint/pull/2194

Reply:
Thanks for the pointer, looks like I'd need to basically rewrite the entire logic when switching to AstRule for which I don't have the time right now. I'll try to convert it when I have more time. Would you accept it as it is and we improve it later? I could leave a refactor comment ...